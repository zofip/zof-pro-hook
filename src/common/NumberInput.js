import React from 'react';

import { makeStyles } from '@material-ui/core/styles';
import clsx from 'clsx';
import InputAdornment from '@material-ui/core/InputAdornment';
import TextField from '@material-ui/core/TextField';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  margin: {
    margin: theme.spacing(1),
  },
  withoutLabel: {
    marginTop: theme.spacing(3),
  },
  textField: {
    flexBasis: 200,
  },
}));


const NumberInput = ({ handleInputValueChange }) => {

  const classes = useStyles();

  return (<TextField
    className={clsx(classes.margin, classes.textField)}
    onChange={handleInputValueChange}
    InputProps={{ startAdornment: <InputAdornment position="start">#</InputAdornment>, }} />
  );
}

export default NumberInput;