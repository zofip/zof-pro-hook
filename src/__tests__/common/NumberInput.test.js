import React from 'react';
import { shallow } from 'enzyme';
import NumberInput from '../../common/NumberInput';

describe('<NumberInput />', () => {
   it('Rendering', () => {
      const numberInput = shallow(<NumberInput />);
      expect(numberInput).toMatchSnapshot();
   });
});