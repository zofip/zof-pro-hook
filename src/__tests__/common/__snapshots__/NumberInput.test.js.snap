// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberInput /> Rendering 1`] = `
<WithStyles(ForwardRef(TextField))
  InputProps={
    Object {
      "startAdornment": <ForwardRef(WithStyles)
        position="start"
      >
        #
      </ForwardRef(WithStyles)>,
    }
  }
  className="makeStyles-margin-2 makeStyles-textField-4"
/>
`;
